android/src/main/java/com/swmansion/rnscreens/ScreenViewManager.kt

```diff
@@ -35,6 +37,18 @@ class ScreenViewManager : ViewGroupManager<Screen>(), RNSScreenManagerInterface<
         setActivityState(view, activityState.toInt())
     }
 
+    override fun updateState(
+        view: Screen,
+        props: ReactStylesDiffMap?,
+        stateWrapper: StateWrapper?
+    ): Any? {
+        if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED && view.fabricViewStateManager != null) {
+            // fabricViewStateManager should never be null in Fabric. The null check is only for Paper's empty impl.
+            view.fabricViewStateManager!!.setStateWrapper(stateWrapper)
+        }
```

Comment:
Actually, after some time of thinking I think it would be better to change this to:

```suggestion
        if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
            // fabricViewStateManager should never be null in Fabric. The null check is only for Paper's empty impl.
            view.fabricViewStateManager?.setStateWrapper(stateWrapper)
        }
```

sorry that I didn't mentioned it above 🤞